homeassistant/helpers/device_registry.py

```diff
@@ -475,6 +480,7 @@ def __init__(self, hass: HomeAssistant) -> None:
             atomic_writes=True,
             minor_version=STORAGE_VERSION_MINOR,
         )
+        self._normalized_name_device_idx: dict[str, list[str]] = {}
```

Comment:
Usually we would put the index in `DeviceRegistryItems`  See the entity_registry for an example

Reply:
My bad, I simply copied the functionality from an [old PR](https://github.com/home-assistant/core/pull/101245) and failed to notice the entire paradigm has changed since then.

[This](https://github.com/tetele/ha-core/blob/25dc34eba0385d337a6b2b6dc25e8def2e705d74/homeassistant/helpers/entity_registry.py#L791) is what I've found. Apparently, there is no search by name in the `entity_registry`. I have [found something](https://github.com/tetele/ha-core/blob/25dc34eba0385d337a6b2b6dc25e8def2e705d74/homeassistant/helpers/area_registry.py#L117-L118) in the `area_registry`, though.

Do you mean I should simply move the index part in the `DeviceRegistryItems` class or that I should leverage the already existing [`DeviceRegistryItems.get_entry`](https://github.com/tetele/ha-core/blob/25dc34eba0385d337a6b2b6dc25e8def2e705d74/homeassistant/helpers/device_registry.py#L448).